dev/fattest.simplicity/src/componenttest/rules/repeater/RepeatTests.java

```diff
@@ -107,9 +110,17 @@ public RepeatTests onlyIf(Supplier<Boolean> check) {
         return this;
     }
 
+    /**
+     * Before each repeat is run the passed in supplier will be run, if it returns false the repeat will be skipped
+     */
+    public RepeatTests additionalEvaluation(Supplier<Boolean> check) {
+        additionalEvaluationCheck = Optional.of(check);
+        return this;
+    }
```

Comment:
1. For primitive types, there are specialized variations of the standard functional interfaces which avoid autoboxing, in this case `BooleanSupplier`.
2. Rather than `BooleanSupplier`, should we use a `Predicate<RepeatTestAction>` here? Usually you'll want to use the action to determine whether to run it and being passed the action directly makes more sense than retrieving the current action.

Reply:
3. If we did that, maybe we should rename the method to `withPredicate`.